src/System.Security.Cryptography.Primitives/src/System/Security/Cryptography/ICryptoTransform.cs

```diff
@@ -26,7 +26,7 @@ public interface ICryptoTransform : IDisposable
         // The return value of TransformBlock is the number of bytes returned to outputBuffer and is
         // always <= OutputBlockSize.  If CanTransformMultipleBlocks is true, then inputCount may be
         // any positive multiple of InputBlockSize
-        int TransformBlock(byte[] inputBuffer, int inputOffset, int inputCount, byte[] outputBuffer, int outputOffset);
+        int TransformBlock(byte[] inputBuffer, int inputOffset, int inputCount, byte[]? outputBuffer, int outputOffset);
```

Comment:
At the interface, `outputBuffer` is non-null.  HashAlgorithm (and derived types) are weird in that they permit null.

Reply:
You mean intent is non null so interface and implementations should be non null?

Reply:
The interface method should say that `outputBuffer` cannot be null (`byte[] outputBuffer`).  The implementations on HashAlgorithm and HashAlgorithm-derived types (which are, I believe, the only implementations in this assembly) can say in their implementations that it can be null (`byte[]? outputBuffer`), since those particular variations permit outputBuffer to be null.